jobs/build/build-sync/Jenkinsfile

```diff
@@ -110,89 +109,152 @@ node {
+        ]
+        if (params.DRY_RUN) {
+            cmd << "--dry-run"
+        }
+        cmd += [
+            "build-sync",
+            "--version=${params.BUILD_VERSION}",
+            "--assembly=${params.ASSEMBLY}"
+        ]
+        if (params.PUBLISH) {
+            cmd << "--publish"
+        }
+        cmd << "--data-path=${params.DOOZER_DATA_PATH}"
+        if (params.EMERGENCY_IGNORE_ISSUES) {
+            cmd << "--emergency-ignore-issues"
+        }
+        if (params.EMERGENCY_IGNORE_ISSUES) {
+            cmd << "--emergency-ignore-issues"
+        }
         if (params.RETRIGGER_CURRENT_NIGHTLY) {
-            if (params.ASSEMBLY != "stream") {
-                error "Cannot use with assembly other than stream. Exiting."
-            }
-            if (params.DRY_RUN) {
-                echo "Would have triggered new release cut in release controller."
-            } else {
-                echo "Triggering release controller to cut new release using previously synced builds..."
-                buildlib.oc("--kubeconfig ${buildlib.ciKubeconfig} -n ocp tag registry.access.redhat.com/ubi8 ${params.BUILD_VERSION}-art-latest:trigger-release-controller")
-                echo "Sleeping so that release controller has time to react..."
-                sleep(60)
-                buildlib.oc("--kubeconfig ${buildlib.ciKubeconfig} -n ocp tag ${params.BUILD_VERSION}-art-latest:trigger-release-controller -d")
-            }
-            return
+            cmd << "--retrigger-current-nightly"
+        }
+        if (params.DOOZER_DATA_GITREF) {
+            cmd << "--doozer-data-gitref=${params.DOOZER_DATA_GITREF}"
+        }
+        if (params.DEBUG) {
+            cmd << "--debug"
+        }
+        if (params.IMAGES) {
+            cmd << "--images=${params.IMAGES}"
+        }
+        if (params.EXCLUDE_ARCHES) {
+            cmd << "--exclude-arches=${params.EXCLUDE_ARCHES}"
+        }
+        if (params.SKIP_MULTI_ARCH_PAYLOAD) {
+            cmd << "--skip-multiarch-payload"
         }
 
-        // // An incident where a bug in oc destroyed the content of a critical imagestream ocp:is/release uncovered the fact that this vital data was not being backed up by any process.
-        // DPTP will be asked to backup etcd on this cluster, but ART should also begin backing up these imagestreams during normal operations as a first line of defense.
-        // In the build-sync job, prior to updating the 4.x-art-latest imagestreams, a copy of all imagestreams in the various release controller namespaces should be performed.
-        stage("backup imagestreams") { build.backupAllImageStreams() }
+        // Run pipeline
+        echo "Will run ${cmd}"
 
-        // This stage is safe to run concurrently. Each build runs
-        // these steps in its own directory.
-        stage("update nightly imagestreams") { build.buildSyncGenInputs() }
+        try {
+            def env = ["KUBECONFIG=${buildlib.ciKubeconfig}"]
```

Comment:
`buildlib.ciKubeconfig` has been removed with https://github.com/openshift/aos-cd-jobs/pull/3335. Needs to use `buildlib.withAppCiAsArtPublish()`

Reply:
Rebased, see [here](https://github.com/openshift/aos-cd-jobs/pull/3324/files#diff-93b849c11dfdf1f70d1625792920d17cbc148073b25e4e0941d14745bd1388eaR200)